### ISSUE TITLE
Can code first be used in other languages？

### ISSUE THREAD

**Issue by GreenShadeZhang** [protobuf-net/protobuf-net.Grpc#158]
When I create a code first service with C #, can I use other languages to call the code first service according to the defined interface?

**Comment by mgravell** [comment]
Yes, but first you must generate the matching .proto schema for your
service; this can *mostly* be automated - but it isn't quite as
well-developed as the code-first bits, purely as a matter of time and
priorities. The SchemaGenerator type in protobuf-net.Grpc.Reflection does
this. I need to improve the samples in this area.

**Comment by mehmetakbulut** [comment]
I currently do this so clients can be written in other languages.

Where I found difficulty is getting schema for multiple services.

I setup a project that uses `SchemaGenerator` to generate schema for each service individually and then merge them while stripping duplicate definitions and statements. There is quite a bit of string parsing and manipulation so it is a bit fragile as you can imagine.

Is there a better way of achieving a single proto file for multiple services?

**Comment by mgravell** [comment]
Currently "no", but in theory it is possible - we did the same thing in
pb-net to the GetSchema API so that multiple types could be passed in. We
could do something very similar, in theory.
